Common/src/containers/CLookUpTable.cpp

```diff
@@ -0,0 +1,488 @@
+    const unsigned long iPoint = edges[iEdge][0];
+    const unsigned long jPoint = edges[iEdge][1];
+
+    /* Loop over all neighobring elements to iPoint. */
+    for (unsigned long iElem = 0; iElem < neighborElemsOfPoint[iPoint].size(); iElem++) {
+      /* loop over 3 points per triangle */
+      for (unsigned long kPoint = 0; kPoint < N_POINTS_TRIANGLE; kPoint++) {
+        /* Get the global ID of the current point. */
+        const unsigned long GlobalIndex = triangles[neighborElemsOfPoint[iPoint][iElem]][kPoint];
+
+        /* Add the current element ID to the neighbor list for this point. */
+        if (GlobalIndex == jPoint) edge_to_triangle[iEdge].push_back(neighborElemsOfPoint[iPoint][iElem]);
+      }
+    }
+  }
+}
+
+void CLookUpTable::ComputeInterpCoeffs(const string& name_prog, const string& name_enth) {
+  /* build KD tree for enthalpy, progress variable space */
+
+  std::array<unsigned long, 3> next_triangle;
+
+  const su2double* prog = GetDataP(name_prog);
+  const su2double* enth = GetDataP(name_enth);
+
+  /* calculate weights for each triangle (basically a distance function) and
+   * build inverse interpolation matrices */
+  for (unsigned long i_triangle = 0; i_triangle < n_triangles; i_triangle++) {
+
+    for (int p = 0; p < 3; p++) {
+      next_triangle[p] = triangles[i_triangle][p];
+    }
+
+    su2activematrix prog_interp_mat_inv(3, 3);
+    GetInterpMatInv(prog, enth, next_triangle, prog_interp_mat_inv);
+    interp_mat_inv_prog_enth.push_back(prog_interp_mat_inv);
+  }
+}
+
+void CLookUpTable::GetInterpMatInv(const su2double* vec_x, const su2double* vec_y,
+                                   std::array<unsigned long,3>& point_ids, su2activematrix& interp_mat_inv) {
+  const unsigned int M = 3;
+  CSquareMatrixCM global_M(3);
+
+  /* setup LHM matrix for the interpolation */
+  for (unsigned int i_point = 0; i_point < M; i_point++) {
+    su2double x = vec_x[point_ids[i_point]];
+    su2double y = vec_y[point_ids[i_point]];
+
+    global_M(i_point,0) = SU2_TYPE::GetValue(1.0);
+    global_M(i_point,1) = SU2_TYPE::GetValue(x);
+    global_M(i_point,2) = SU2_TYPE::GetValue(y);
+  }
+
+  global_M.Invert();
+  global_M.Transpose();
+ 
+  for (unsigned int i=0; i<M; i++){
+    for (unsigned int j=0; j<M; j++){
+      interp_mat_inv[i][j] = global_M(i,j);
+    }
+  }
+
+}
+
+unsigned long CLookUpTable::LookUp_ProgEnth(const string& val_name_var, su2double *val_var, su2double val_prog,
+                                            su2double val_enth, const string& name_prog, const string& name_enth) {
+  unsigned long exit_code = 0;
+
+  if (val_name_var.compare("NULL") == 0) {
+    *val_var = 0.0;
+    exit_code = 0;
+    return exit_code;
+  }
+
+  /* check if progress variable and enthalpy value is in table range */
+  if (val_prog >= limits_table_prog[0] && val_prog <= limits_table_prog[1]) {
+    //&&
+    // val_enth >= limits_table_enth[0] && val_enth <= limits_table_enth[1] ){
```

Comment:
Remove if not needed

Reply:
OK, I think it's actually better to include this check so I've uncommented it.